### ISSUE TITLE
Error for udt library in linking

### ISSUE THREAD

**Issue by vandanachadha** [Haivision/srt#381]
I am trying to create a variant of stransmit and including a modified clone of it in my code. Header files included are srt.h and udt.h, and it compiles. However, at the time of linking, it does not find the library udt.h though it exists in the same location as srt.h and the path has been added to the cmake file -> include_directories command. The errors I get are for example "undefined reference to UriParser::UriParser(...."

If I remove the inclusion of <udt.h> from my cpp file, then I get errors on all stransmit functions like UDT::setlogstream() and all such functions.

Please clarify how can I include the udt.h to my cmake file.

**Comment by ethouris** [comment]
Ok, some information. The `udt.h` header is currently a "legacy and C++ extras" header (`srt.h` is a C-only header containing all required API).

`UriParser` is defined in `uriparser.hpp` file and you need `uriparser.cpp` added to the application. All these things are contained in the `apps` directory. This directory contains single application source files with all others that are shared between applications, so-called "SRT application support".

There are several macros added to simplify the definition of `srt-live-transmit` application in cmake file:

    	srt_add_application(srt-live-transmit ${VIRTUAL_srtsupport})

There's used first something that can be called a "virtual library" (a "library" which is defined in the build file only and "adding" this library to an executable file means to add all source files of this "virtual library" to the source list of the executable). This is done by adding the *.o files instead of source files directly, so `VIRTUAL_srtsupport` lists these *.o files that will be produced out of source files coming from `apps/support.maf` file.

This above `srt_add_application` defines an executable containing as sources the file named the same as the application (with `.cpp`) and other sources passed as the next argument. Then it adds appropriate include directories and linker options, as needed - it's best to trace how this is defined inside.

In short: the `srt-live-transmit` application consists of:
* `apps/srt-live-transmit.cpp`
* files which's names are collected in `apps/support.maf` file
and it should include directories in `apps` and `srtcore`, then link against SRT library.
